bftengine/src/bftengine/KeyExchangeManager.cpp

```diff
@@ -245,7 +248,7 @@ void KeyExchangeManager::generateConsensusKeyAndSendInternalClientMsg(const SeqN
   }
 
   auto& candidate = candidate_private_keys_.generated;
-  bool generateNewPair = (sn != candidate.sn) || candidate.cid.empty();
+  bool generateNewPair = ((sn != candidate.sn) || candidate.cid.empty()) && (seq_candidate_map_.count(sn) == 0);
```

Comment:
once you `seq_candidate_map_.count(sn) == 0`, you don't need to check `sn != candidate.sn`, right?

Reply:
yes correct, if candidate is not there in map, then sn will never be equal to candidate.sn. I will remove this check.